workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Partials/Improbable.Coordinates

```diff
@@ -0,0 +1,166 @@
+}
+
+public static Coordinates operator -(Coordinates v1, Vector3f v2)
+{
+    return new Coordinates(v1.X - v2.X, v1.Y - v2.Y, v1.Z - v2.Z);
+}
+
+public static Coordinates operator +(Coordinates a, Vector3f b)
+{
+    return new Coordinates(a.X + b.X, a.Y + b.Y, a.Z + b.Z);
+}
+
+public static Coordinates operator -(Coordinates v1, Vector3 v2)
+{
+    return new Coordinates(v1.X - v2.x, v1.Y - v2.y, v1.Z - v2.z);
+}
+
+public static Coordinates operator +(Coordinates a, Vector3 b)
+{
+    return new Coordinates(a.X + b.x, a.Y + b.y, a.Z + b.z);
+}
+
+public static Coordinates operator -(Vector3d v1, Coordinates v2)
+{
+    return new Coordinates(v1.X - v2.X, v1.Y - v2.Y, v1.Z - v2.Z);
+}
+
+public static Coordinates operator -(Vector3f v1, Coordinates v2)
+{
+    return new Coordinates(v1.X - v2.X, v1.Y - v2.Y, v1.Z - v2.Z);
+}
+
+public static Coordinates operator -(Vector3 v1, Coordinates v2)
+{
+    return new Coordinates(v1.x - v2.X, v1.y - v2.Y, v1.z - v2.Z);
+}
+
+public static Coordinates operator +(Vector3d a, Coordinates b)
+{
+    return new Coordinates(a.X + b.X, a.Y + b.Y, a.Z + b.Z);
+}
+
+public static Coordinates operator +(Vector3f a, Coordinates b)
+{
+    return new Coordinates(a.X + b.X, a.Y + b.Y, a.Z + b.Z);
+}
+
+public static Coordinates operator +(Vector3 a, Coordinates b)
+{
+    return new Coordinates(a.x + b.X, a.y + b.Y, a.z + b.Z);
+}
+
+public static bool operator ==(Coordinates a, Coordinates b) => a.Equals(b);
+public static bool operator !=(Coordinates a, Coordinates b) => !a.Equals(b);
+
+public override bool Equals(object obj) 
+{
+    if (obj is Coordinates == false) 
+    {
+        return false;
+    }
+
+    var otherCoord = (Coordinates)obj;
+    return X == otherCoord.X && Y == otherCoord.Y && Z == otherCoord.Z;
+}
+
+/// <summary>
+///     Interpolates between to global coordinates
+/// </summary>
+/// <param name="currentPosition">The starting position</param>
+/// <param name="newPosition">The position to intepolate towards</param>
+/// <param name="progressRatio">
+///     where 0 is the currentPosition and 1 is newPosition. Note: the value is clamped between 0
+///     and 1 to prevent extrapolation.
+/// </param>
+/// <returns>The interpolated position</returns>
+public static Coordinates Lerp(Coordinates currentPosition, Coordinates newPosition, float progressRatio)
+{
+    progressRatio = progressRatio > 1 ? 1 : (progressRatio < 0 ? 0 : progressRatio); // Clamp
+    var valueDelta = newPosition - currentPosition;
+    return valueDelta * progressRatio + currentPosition;
+}
+
+/// <summary>
+///     Check if a coordinate is near another.
+/// </summary>
+/// <param name="other">The coordinate to test.</param>
+/// <param name="distance">The allowed range.</param>
+/// <returns>True if the other coordinate is within strictly less than the specified range.</returns>
+public bool IsWithinDistance(Coordinates other, double distance)
+{
+#pragma warning disable 618
```

Comment:
Why is this warning disable necessary?

Reply:
Not sure if it is. This is all swiped from the SDK. This appears to be related to the `Obsolete` attribute 🤔 https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0618

Will remove and see what happens

Reply:
Removed